lib_test/test_h2_server.ml

```diff
@@ -1189,6 +1189,70 @@ module Server_connection_tests = struct
     | _ ->
       assert false
 
+  let trailers_request_handler reqd =
+    let response = Response.create `OK in
+    (* Send the response for / *)
+    let response_body = Reqd.respond_with_streaming reqd response in
+    Body.write_string response_body "somedata";
+    Body.flush response_body (fun () ->
+        Reqd.send_trailers_on_close reqd Headers.(add empty "foo" "bar");
+        Body.close_writer response_body)
+
+  let test_trailers () =
+    let t = create ~error_handler trailers_request_handler in
+    handle_preface t;
+    let headers, _ = header_and_continuation_frames in
+    let headers =
+      { headers with
+        Frame.frame_header =
+          { headers.frame_header with
+            flags = Flags.(default_flags |> set_end_header |> set_end_stream)
+          }
+      }
+    in
+    read_frames t [ headers ];
+    match next_write_operation t with
+    | `Write iovecs ->
+      let frames = parse_frames (Write_operation.iovecs_to_string iovecs) in
+      Alcotest.(check (list int))
+        "Next write operation surfaces writes for HEADERS / DATA"
+        List.(map Frame.FrameType.serialize Frame.FrameType.[ Headers; Data ])
```

Comment:
it would be cool to verify that this `DATA` frame doesn't have the `END_STREAM` flag set.